apps/src/studio/studio.js

```diff
@@ -858,10 +869,24 @@ Studio.onTick = function() {
     performItemOrProjectileMoves(Studio.items);
   }
 
+  Studio.updateFloatingScore();
+
   sortDrawOrder();
 
-  if (!spritesNeedMoreAnimationFrames && checkFinished()) {
-    Studio.onPuzzleComplete();
+  if (!Studio.succeeded && checkFinished()) {
+    Studio.succeeded = true;
+    Studio.succeededTime = new Date().getTime();
+  }
+
+  if (Studio.succeeded) {
+    if (level.delayCompletion) {
```

Comment:
I wonder if these could be combined differently.
`if (!spritesNeedMoreAnimationFrames && (!level.delayCompletion || currentTime > Studio.succeededTime))`